### PR TITLE
Send explosions only to nearby players

### PR DESCRIPTION
**Pull request by joelramilison**
## Proposed changes

Explosion packets are now only sent to nearby players to improve performance.

## Types of changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of
them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before
merging your code._

- [x] I have read the CONTRIBUTING.md
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

Fixes issue #64

### CODE REVIEW

**Comment by TheMeinerLP** [comment]
Can you try to re-implement also a test for that behaviour ?

**Review by OneLiteFeather — APPROVED** [review]
Looks good to me

**Comment by OneLiteFeather** [comment]
> Can you try to re-implement also a test for that behaviour ?

That would be beneficial to do so, I would also welcome that if @joelramilison wants to add this

**Comment by TheMeinerLP** [comment]
> Looks good to me

The problem is, we can't approve it because the pipelines fails

**Comment by joelramilison** [comment]
What would a unit test for that look like, what are you thinking of? I'm asking because the code changes were quite minimal, the getNearbyEntities function is already being tested and the rest seems quite straightforward.

**Comment by TheMeinerLP** [comment]
Regardless of whether the change is small, testing allows us to ensure that the quality remains the same in version updates. In addition, we can ensure that nothing breaks in the event of changes to other areas that may only have an indirect effect.

File: src/test/java/net/minestom/server/instance/ExplosionIntegrationTest.java
Comment (theEvilReaper) [inline src/test/java/net/minestom/server/instance/ExplosionIntegrationTest.java:12]: This empty line can be removed

File: src/test/java/net/minestom/server/instance/ExplosionIntegrationTest.java
Comment (theEvilReaper) [inline src/test/java/net/minestom/server/instance/ExplosionIntegrationTest.java:41]: It’s recommended to destroy the used instance and players after your test completes. For this purpose, the `Env` class provides a method you can use.

**Review by theEvilReaper — CHANGES_REQUESTED** [review]

File: src/test/java/net/minestom/server/instance/ExplosionIntegrationTest.java
Comment (theEvilReaper) [inline src/test/java/net/minestom/server/instance/ExplosionIntegrationTest.java]: In JUnit 5, it's not necessary to use the `public` keyword for class declarations or test methods. You can remove it from your test classes and methods.

In Chapter [2.3](https://junit.org/junit5/docs/current/user-guide/), you can find some information about that behaviour

Comment (joelramilison) [reply src/test/java/net/minestom/server/instance/ExplosionIntegrationTest.java]: Thank you for the valuable feedback! I've pushed the changes.

**Comment by TheMeinerLP** [comment]
Can you resolve the merge conflicts ?

**Review by TheMeinerLP — APPROVED** [review]

**Review by theEvilReaper — APPROVED** [review]
LGTM

**Comment by TheMeinerLP** [comment]
We need to test the tests manually before we merge

**Comment by TheMeinerLP** [comment]
image
Works as planned

**Pull request merged**
